api/vscode-azureresources-api.d.ts

```diff
@@ -0,0 +1,417 @@
+import type { Activity } from '@microsoft/vscode-azext-utils/hostapi';

```

Comment:
Is it possible to cause comments to be written at the top of this file? I'm thinking:
1. Copyright header
2. A note that it is a generated file and manual changes would be overwritten
3. Instructions on how to generate it

Though, would it be more appropriate to generate it only for the build, and not check it in?

Reply:
> Is it possible to cause comments to be written at the top of this file? I'm thinking:
> 
> 1. Copyright header
> 2. A note that it is a generated file and manual changes would be overwritten
> 3. Instructions on how to generate it

Yes, adding a comment or at the very least a readme with those is needed. Sadly it doesn't seem possible to add a comment using api-extractor alone. But we could add a script to do it.

> Though, would it be more appropriate to generate it only for the build, and not check it in?

I thought about this and I think it's best to check it in:
1. I want to be able to track changes to it
2. In case api-extractor does anything weird or anything we don't like we can catch it

I'm open to generating it at build if you think it's better.

Reply:
I don't feel strongly about it and I think those are good reasons. One counterpoint is that, if we _don't_ pack only the built version, then the checked-in file could get out of sync with what actually is built (whether we rerun api-extractor every build or not, either way what's checked in could get out-of-sync with what's build). We can certainly try to enforce it at PR time but the potential for mistakes is definitely there; I feel like we make that sort of mistake pretty regularly in the `@microsoft/vscode-azext-utils` package. 

RE: copyright header, it stinks we'd have to go as far as writing a script to update this file with those things, it seems like an obvious feature for api-extractor to have a config for.

Reply:
> RE: copyright header, it stinks we'd have to go as far as writing a script to update this file with those things, it seems like an obvious feature for api-extractor to have a config for.

https://github.com/microsoft/rushstack/issues/2584

Reply:
> I don't feel strongly about it and I think those are good reasons. One counterpoint is that, if we _don't_ pack only the built version, then the checked-in file could get out of sync with what actually is built (whether we rerun api-extractor every build or not, either way what's checked in could get out-of-sync with what's built). We can certainly try to enforce it at PR time but the potential for mistakes is definitely there; I feel like we make that sort of mistake pretty regularly in the `@microsoft/vscode-azext-utils` package.

Ok, I think I'm convinced doing it at build time (npm pack time) is best.

Reply:
However, I might still want to check it in purely for reference and documentation. (I like to be able to link to it on GitHub).